File: Python/5kyu/Weight for Weight/solution.py
# O(N²), not good :/

def order_weight(strng):
    numbers = strng.split();
    sum = 0
    sum_arr = []
    
    for number in numbers:
        for num in number:
            sum += int(num)
        sum_arr.append(sum)
        sum = 0

    z = [x for _, x in sorted(zip(sum_arr, numbers))]
    return " ".join(z)

File: PHP/5kyu/Weight for Weight/solution.php
<?php
function orderWeight($str) {
  $nums = explode(" ", $str);
  $sumArr = array();
  $sum = 0;
  foreach($nums as $num) {
    for($i = 0; $i < strlen($num); $i++)
      $sum += strval($num[$i]);

    array_push($sumArr, $sum);
    $sum = 0;
  }

  array_multisort($sumArr, $nums, SORT_STRING);
  return implode(' ', $nums);
}
?>